src/components/GoalActivityFeed.tsx

```diff
@@ -0,0 +1,191 @@
+import { GoalByIdReturnType } from '../../trpc/inferredTypes';
+import { useGoalResource } from '../hooks/useGoalResource';
+
+import { GoalDeleteModal } from './GoalDeleteModal/GoalDeleteModal';
+import { CommentView } from './CommentView/CommentView';
+import { GoalDependencyAddForm } from './GoalDependencyForm/GoalDependencyForm';
+import { GoalDependencyListByKind } from './GoalDependencyList/GoalDependencyList';
+import { GoalCriteria } from './GoalCriteria/GoalCriteria';
+import { GoalActivity } from './GoalActivity';
+import { AddCriteriaForm } from './CriteriaForm/CriteriaForm';
+
+const ModalOnEvent = dynamic(() => import('./ModalOnEvent'));
+const GoalEditForm = dynamic(() => import('./GoalEditForm/GoalEditForm'));
+const CommentCreateForm = dynamic(() => import('./CommentCreateForm/CommentCreateForm'));
+
+type GoalActivityFeedProps = {
+    goal?: GoalByIdReturnType;
+    shortId?: string;
+    activityId?: string;
+
+    onGoalCriteriaClick?: ComponentProps<typeof GoalCriteria>['onClick'];
+    onGoalDependencyClick?: ComponentProps<typeof GoalDependencyListByKind>['onClick'];
+    onDelete?: () => void;
+};
+
+export const GoalActivityFeed = forwardRef<HTMLDivElement, GoalActivityFeedProps>(
+    ({ goal, shortId, activityId, onGoalCriteriaClick, onGoalDependencyClick, onDelete }, ref) => {
+        const {
+            onGoalCommentUpdate,
+            onGoalDelete,
+            onGoalCriteriaAdd,
+            onGoalCriteriaToggle,
+            onGoalCriteriaUpdate,
+            onGoalCriteriaRemove,
+            onGoalCriteriaConvert,
+            resolveGoalCommentDraft,
+            onGoalCommentChange,
+            onGoalCommentCreate,
+            onGoalCommentCancel,
+            onGoalCommentReactionToggle,
+            onGoalCommentDelete,
+            onGoalDependencyAdd,
+            onGoalDependencyRemove,
+            lastStateComment,
+            highlightCommentId,
+        } = useGoalResource(
+            {
+                id: goal?.id,
+                stateId: goal?.stateId,
+                reactions: goal?.reactions,
+                comments: goal?.comments,
+            },
+            {
+                invalidate: {
+                    getById: shortId,
+                },
+            },
+        );
+
+        const commentDraft = resolveGoalCommentDraft(goal?.id);
+
+        const onGoalDeleteConfirm = useCallback(() => {
+            onGoalDelete();
+            onDelete?.();
+        }, [onGoalDelete, onDelete]);
+
+        const onGoalCommentSubmit = useCallback(
+            (value: NonNullable<GoalByIdReturnType>['comments'][number]) => {
+                return value.activity?.id === activityId ? onGoalCommentUpdate(value.id) : undefined;
+            },
+            [onGoalCommentUpdate, activityId],
+        );
+
+        return (
+            <>
+                {nullable(goal, ({ _activityFeed, id, goalAchiveCriteria, _relations, project, _isEditable }) => (
```

Comment:
This logic must be in parent component. U can check it in parent and make goal prop required to render this component